src/app/sub.rs

```diff
@@ -69,6 +71,29 @@ where
             no_sort,
         } = self;
 
+        // If one of the files is on HDD, set thread number to 1
+        let disks = Disks::new_with_refreshed_list();
+
+        if files
+            .iter()
+            .filter_map(|file| fs::canonicalize(file).ok())
+            .any(|path| {
+                let mount_points: Vec<_> = disks.iter().map(|disk| disk.mount_point()).collect();
```

Comment:
Collecting mount points into a `Vec` is suboptimal if `files` has only 1 item. We should only collect only if `files.len() > 1`.

You can change `find_mount_point` into taking an `impl IntoIterator`.